### PR TITLE
fix(repo): expand URIs into valid deb repositories

### LINKED ISSUE #75: [20.10] "Add Source" dialog hangs when attempting to add invalid source
In 20.04, if I try to add an invalid source (using `http://test` as an example), I get a "Couldn't add source" dialog and the source is then added anyway (it's not good that the action is contrary to the error message, but that's sort of covered under this other issue: https://github.com/pop-os/repoman/issues/68).

In 20.10, if I try to add the same invalid source, instead of displaying an error message, the "Add Source" dialog box just hangs on its spinner indefinitely. Both the "Add" and "Cancel" buttons are greyed out, so I'm not able to back out or cancel without killing the program.

repoman-issue-screenshot1

While I don't expect users to intentionally add bad source lines, this could happen in the event of a bad/incomplete copy/paste, and I would expect it to time out eventually so I can try again.

### PR DESCRIPTION
**Pull request by isantop**
Ensures that even if a user enters a URI, it will expand to a valid repository.

Fixes #75

### CODE REVIEW

**Review by leviport — DISMISSED** [review]
Looks good on Focal and Groovy

File: repoman/repo.py [hunk 1]
```diff
@@ -172,11 +172,21 @@ def get_os_name():
     return "your OS"
 
 def _do_add_source(name, line, dialog):
+    enable = True
```
Comment (crawfxrd) [inline repoman/repo.py]: Is this value used for anything? I see it set, but not where it's used.

Comment (isantop) [reply repoman/repo.py]: fixed

**Review by jacobgkau — APPROVED** [review]
Still working as expected (and fixes https://github.com/pop-os/repoman/issues/75) after the last change.

**Review by ids1024 — APPROVED** [review]

**Pull request merged**